codalab/rest/help.py

```
@@ -0,0 +1,30 @@
+import httplib
+
+from bottle import abort, get, request, local, post, template
+
+from codalab.lib.spec_util import NAME_REGEX
+from codalab.lib.server_util import bottle_patch as patch
+from codalab.rest.schemas import (
+    AuthenticatedUserSchema,
+    USER_READ_ONLY_FIELDS,
+    UserSchema,
+)
+from codalab.server.authenticated_plugin import (
+    AuthenticatedPlugin,
+    UserVerifiedPlugin,
+)
+
+@post('/help/', apply=AuthenticatedPlugin(), skip=UserVerifiedPlugin)
+def fetch_help():
+    support_email = local.config['support_email']
```

What should happen if 'support_email' isn't defined in the config? Or should we require that 'support_email' always be defined?

---

How have we dealt with similar kinds of configuration issues? I think we should require that it's always defined.

---

Usually we try to provide a sane default when something is not defined, or just let it through with a warning message, unless it makes no sense to do so.

Example with the `admin_email` config:
https://github.com/codalab/codalab-cli/blob/master/codalab/server/rest_server.py#L200-L205

The issue is that requiring the config may cause a deployment to crash if we just forgot to update the config on the server. In this case it should be pretty isolated and not cause much problems though. Maybe we can just let this throw an exception to let us know we forgot to configure a support email, or throw your own exception to give a more helpful error message.

---

Ok, well why don't I just follow the pattern we've already established, and I'll just print off a message saying that the support_email needs to be configured, and then print of the email I would've sent to the console.

---

Ok, changes made!